packages/pwa-kit-dev/bin/pwa-kit-dev.js

```diff
@@ -285,12 +285,16 @@ const main = async () => {
                 credentialsFile
             }) => {
                 // Set the deployment target env var, this is required to ensure we
-                // get the correct configuration object.
-                process.env.DEPLOY_TARGET = target
+                // get the correct configuration object. Do not assign the variable it if
+                // the target value is `undefined` as it will serialied as a "undefined"
+                // string value.
+                if (target) {
+                    process.env.DEPLOY_TARGET = target
+                }
```

Comment:
I noticed that when this environment variable is used in the `ssr-config.js` file, its value was always defined. In fact it was a string literal "undefined", this meant that the locations were cosmiconfig searched included `config/undefined.ext` meaning you could drop a file named "undefined.json" in your config and it would try reading it. 

Adding a guard fixed this.